### PR TITLE
Updates to vignettes and pkgdown

### PR DESCRIPTION
**Pull request by joshwlambert**
This PR addresses:
- #139 
- #140 
- #143 

Update vignettes:
- now using CSL-JSON to handle bibliography
- link citations 

Improved pkgdown site:
- Add custom function reference to group functions
- Add custom article reference to group and navbar header to vignettes
- Turn on code folding in vignettes (default show code, except for plotting chunks)

Removed functions from reference:
-  `create_prob_dist()` (exported -> unexported)
- `check_optim_conv()` (unexported in manual -> unexported not in manual)

### CODE REVIEW

**Comment by joshwlambert** [comment]
Ideally, `ends_with()` or `matches()` would be used for the `<epidist>` and `<vb_epidist>` methods in the pkgdown reference. However, the distributions functions are methods of these classes and they are separated under their own header in the pkgdown reference. For this reason only `<epiparam>` uses `ends_with()` to group class methods.

File: _pkgdown.yml [hunk 1]
```diff
@@ -11,4 +13,127 @@ template:
...
+    contents:
+    - cdf.epidist
+
+  - title: Parameter extraction
+    desc: Extract distribution parameters from summary statistics
+    contents:
+    - extract_param
+
+  - title: Parameter conversion
+    desc: Converts distribution parameter to summary staistics and vice versa
```
Comment (Bisaloo) [inline _pkgdown.yml:78]:
```suggestion
    desc: Converts distribution parameter to summary statistics and vice versa
```

**Comment by Bisaloo** [comment]
> Ideally, ends_with() or matches() would be used for the <epidist> and <vb_epidist> methods in the pkgdown reference. However, the distributions functions are methods of these classes and they are separated under their own header in the pkgdown reference. For this reason only <epiparam> uses ends_with() to group class methods.

You can create complex selection combinations in pkgdown index. E.g., `ends_with("\\.epidist)` and not tagged with with a specific keyword.

For example:

https://github.com/epiverse-trace/linelist/commit/372d2887432cc836b66f0149fba128dd5b9a2f21#diff-4696b5e124c0b2a7e0ef4617c2e5be11de4ea153568c38fe79ee4925a8cb6b2e

**Comment by joshwlambert** [comment]
@Bisaloo Thanks for the tip, I didn't realise this was possible. Updates applied in a1cf03d6a22989ba7b3bfee4c85fb4126ac01abe

**Comment by jamesmbaazam** [comment]
This PR is very insightful.

**Pull request merged**